### PR TITLE
fix: wrap `fetch_watermarks` in `spawn_blocking`

### PR DESCRIPTION
**Pull request by wangrunji0408**
Please take a look. @BugenZhao @tabVersion

Will release as madsim-rdkafka v0.3.3.

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]
generally LGTM

**Comment by BugenZhao** [comment]
Is there a plan for other synchronous interfaces as well? 🥺

**Comment by wangrunji0408** [comment]
> Is there a plan for other synchronous interfaces as well? 🥺

Maybe migrated case by case. 🥹

File: madsim-rdkafka/src/std/client.rs [hunk 1]
```diff
@@ -354,29 +354,32 @@ impl<C: ClientContext> Client<C> {
...
         partition: i32,
         timeout: T,
     ) -> KafkaResult<(i64, i64)> {
-        let mut low = -1;
-        let mut high = -1;
         let topic_c = CString::new(topic.to_string())?;
-        let ret = unsafe {
-            rdsys::rd_kafka_query_watermark_offsets(
-                self.native_ptr(),
+        let native_client = unsafe { NativeClient::from_ptr(self.native_ptr()) };
```
Comment (BugenZhao) [inline madsim-rdkafka/src/std/client.rs:364]: `NativeClient` will call `rd_kafka_destroy` when it's get dropped (behavior of `NativePtr`). Will this lead to problem?

Comment (wangrunji0408) [reply madsim-rdkafka/src/std/client.rs:364]: Ohhh. I think so. Thanks for your troubleshooting!

**Pull request merged**